pages/[league]/player/[id].tsx

```diff
@@ -0,0 +1,337 @@
+import { GetServerSideProps } from 'next';
+import { NextSeo } from 'next-seo';
+import { useRouter } from 'next/router'
+import React, { useEffect, useState } from 'react';
+import { PulseLoader } from 'react-spinners';
+import styled from 'styled-components';
+
+import Footer from '../../../components/Footer';
+import Header from '../../../components/Header';
+import SeasonTypeSelector from '../../../components/Selector/SeasonTypeSelector';
+import SingleGoalieRatingsTable from '../../../components/SingleRatingsTable/SingleGoalieRatingsTable';
+import SinglePlayerRatingsTable from '../../../components/SingleRatingsTable/SingleSkaterRatingsTable';
+import SingleGoalieScoreTable from '../../../components/SingleScoreTable/SingleGoalieScoreTable';
+import SingleSkaterAdvStatsTable from '../../../components/SingleScoreTable/SingleSkaterAdvStatsTable';
+import SingleSkaterScoreTable from '../../../components/SingleScoreTable/SingleSkaterScoreTable';
+import useGoalieInfo from '../../../hooks/useGoalieInfo';
+import useGoalieRatingsId from '../../../hooks/useGoalieRatingsId';
+import useGoalieStatsId from '../../../hooks/useGoalieStatsId';
+import useRatingsId from '../../../hooks/useRatingsId';
+import useSkaterInfo from '../../../hooks/useSkaterInfo';
+import useSkaterStatsId from '../../../hooks/useSkaterStatsId';
+import { SeasonType } from '../../api/v1/players/stats';
+
+interface Props {
+  league: string;
+  id: number;
+}
+function PlayerPage({ league, id }: Props): JSX.Element {
+
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [isSkater, setIsSkater] = useState<boolean>(true);
+  const [playerError, setPlayerError] = useState<boolean>(false);
+  const [filterSeasonType, setFilterSeasonType] = useState('Regular Season');
+
+  const router = useRouter();
+
+  // player info
+  const [playerName, setPlayerName] = useState('');
+  const [playerPosition, setPlayerPosition] = useState('');
+  const [playerTeam, setPlayerTeam] = useState('');
+  const [playerWeight, setPlayerWeight] = useState('');
+  const [playerHeight, setPlayerHeight] = useState('');
+
+  // ratings
+  const {
+    ratings: goalieRatings,
+    isLoading: isLoadingGoalieRating,
+    isError: isErrorGoalieRating,
+  } = useGoalieRatingsId(id, league);
+
+  const {
+    ratings: skaterRatings,
+    isLoading: isLoadingPlayerRating,
+    isError: isErrorPlayerRating,
+  } = useRatingsId(id, league);
+
+  // stats
+  const {
+    ratings: goalieStats,
+    isLoading: isLoadingGoalieStats,
+    isError: isErrorGoalieStats,
+  } = useGoalieStatsId(id, league, filterSeasonType);
+
+  const {
+    ratings: skaterStats,
+    isLoading: isLoadingSkaterStats,
+    isError: isErrorSkaterStats,
+  } = useSkaterStatsId(id, league, filterSeasonType);
+
+  // player info
+  const {
+    ratings: goalieInfo,
+    isLoading: isLoadingGoalieInfo,
+    isError: isErrorGoalieInfo,
+  } = useGoalieInfo(id, league)
+
```

Comment:
I wonder if we can think of a better way to do this?

Reply:
im all ears